src/persist-client/src/impl/machine.rs

```diff
@@ -758,15 +761,86 @@ mod tests {
 
             let state = Arc::new(Mutex::new(DatadrivenState::default()));
             let cpu_heavy_runtime = Arc::new(CpuHeavyRuntime::new());
+            let write = Arc::new(Mutex::new(
+                client
+                    .open_writer::<String, (), u64, i64>(shard_id)
+                    .await
+                    .expect("invalid shard types"),
+            ));
 
             f.run_async(move |tc| {
                 let shard_id = shard_id.clone();
                 let client = client.clone();
                 let state = Arc::clone(&state);
                 let cpu_heavy_runtime = Arc::clone(&cpu_heavy_runtime);
+                let write = Arc::clone(&write);
                 async move {
                     let mut state = state.lock().await;
+
                     match tc.directive.as_str() {
+                        // Returns whether a blob contains a given batch. The batch
+                        // must contain only a single key.
```

Comment:
musing, quite optional: instead of enforcing a single key, it could print a line per key to the output about whether that key exists. I think that accomplishes the same thing but without the restriction